File: php/iforevents/src/MemoryStorage.php
<?php

declare(strict_types=1);

namespace IForevents;

/** Nothing survives the request; the default. */
final class MemoryStorage implements Storage
{
    /** @var array<string, string> */
    private array $data = [];

    public function get(string $key): ?string
    {
        return $this->data[$key] ?? null;
    }

    public function set(string $key, string $value): void
    {
        $this->data[$key] = $value;
    }

    public function remove(string $key): void
    {
        unset($this->data[$key]);
    }
}

File: java/iforevents/src/main/java/com/iforevents/MemoryStorage.java
package com.iforevents;

import java.util.concurrent.ConcurrentHashMap;

/** Default storage: nothing survives the process. */
public final class MemoryStorage implements Storage {
    private final ConcurrentHashMap<String, String> data = new ConcurrentHashMap<String, String>();

    @Override
    public String get(String key) {
        return data.get(key);
    }

    @Override
    public void set(String key, String value) {
        data.put(key, value);
    }

    @Override
    public void remove(String key) {
        data.remove(key);
    }
}
